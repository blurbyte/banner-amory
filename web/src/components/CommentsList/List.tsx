import styled from 'styled-components';

const List = styled.ul`
  margin: 0;
  margin-top: 1rem;
  padding: 0;
  list-style: 0;
`;

export default List;
