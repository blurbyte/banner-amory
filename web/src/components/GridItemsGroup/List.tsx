// grid containing items under single character

import styled from 'styled-components';

type ListProps = {
  columns: number;
};

const List = styled.ul<ListProps>`
  display: grid;
  grid-template-columns: repeat(${props => props.columns}, 1fr);
  grid-gap: 1rem;
  margin: 0;
  padding: 0;
  list-style: none;
`;

export default List;
