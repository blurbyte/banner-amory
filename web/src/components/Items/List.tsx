// items grid

import * as React from 'react';
import styled from 'styled-components';

import { ItemBasic } from '../../types/Item';
import ListItem from './ListItem';

const Grid = styled.ul`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat(10, 1fr);
  margin: 0;
  padding: 0;
  list-style: none;
`;

type ListProps = {
  items: [ItemBasic];
};

class List extends React.Component<ListProps> {
  render() {
    const { items } = this.props;

    return <Grid>{items.map(item => <ListItem key={item.slug} {...item} />)}</Grid>;
  }
}

export default List;
