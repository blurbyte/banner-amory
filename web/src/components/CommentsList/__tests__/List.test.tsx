import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import List from '../List';

test('renders correctly', () => {
  const component = renderer.create(<List />);
  expect(component.toJSON()).toMatchSnapshot();
});
