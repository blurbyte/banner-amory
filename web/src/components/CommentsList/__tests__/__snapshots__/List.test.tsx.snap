// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  margin: 0;
  margin-top: 1rem;
  padding: 0;
  list-style: 0;
}

<ul
  className="c0"
/>
`;
